### PR TITLE
EZP-30524 Rewrite Universal Discovery Widget

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://jira.ez.no/browse/EZP-30524
| Versions      | master

### CODE REVIEW

File: docs/guide/extending/extending_udw.md [hunk 1]
```diff
@@ -33,120 +29,131 @@ const props = {
 
 ## Adding new tabs to the UDW
 
-The Universal Discovery Widget is highly customizable. It allows you to add new tabs to the module.
+The Universal Discovery Widget is highly customizable.
+It allows you to add new tabs to the module ijn the following way:
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
It enables you to add new tabs to the module in the following way:
```

File: docs/guide/extending/extending_udw.md [hunk 2]
```diff
@@ -33,120 +29,131 @@ const props = {
...
-- **allowedContentTypes** _{Array}_ - Content Types that are available in other UDW tabs,
-- **cotfPreselectedLocation** _{Number}_ - Location that is preselected in Content on the Fly,
-- **cotfAllowedLocations** _{Array}_ - Locations that are available in Content on the Fly
-
-The following props are deprecated:
-
-- **languages** and **contentTypes** are lists of languages and Content Types in the system, read from the application config.
+|Property|Value|Definition|
+|-----------|------|----------|
+|id|string|Tab ID - `browse`|
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
|id|string|Tab ID, e.g. `browse`|
```
I guess?

File: docs/guide/extending/extending_udw.md [hunk 3]
```diff
@@ -33,120 +29,131 @@ const props = {
...
+|component|element|React component that represents content of a tab|
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
|component|element|React component that represents the contents of a tab|
```

File: docs/guide/extending/extending_udw.md [hunk 4]
```diff
@@ -33,120 +29,131 @@ const props = {
...
+|label|string|Label text e.g. `Browse`|
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
|label|string|Label text, e.g. `Browse`|
```

File: docs/guide/extending/extending_udw.md [hunk 5]
```diff
@@ -33,120 +29,131 @@ const props = {
...
+|icon|string|Path to the icon e.g. `/bundles/ezplatformadminui/img/ez-icons.svg#browse`|
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
|icon|string|Path to the icon, e.g. `/bundles/ezplatformadminui/img/ez-icons.svg#browse`|
```

File: docs/guide/extending/extending_udw.md [hunk 6]
```diff
@@ -33,120 +29,131 @@ const props = {
...
 
 ## Configuration
 
 You can configure Universal Discovery Widget in the [`universal_discovery_widget.yaml`](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/src/bundle/Resources/config/universal_discovery_widget.yaml) file.
-There you can set e.g. the starting location ID, visible tabs, allowed Content Types, search limits, etc.
+
+There you can set following properties:
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
There you can set the following properties:
```

File: docs/guide/extending/extending_udw.md [hunk 7]
```diff
@@ -33,120 +29,131 @@ const props = {
...
+
+|YML</br>React props|Values|Required|Definition|
+|-------------------|------|--------|----------|
+|multiple</br>`multiple`|true</br>false|no|The possibility to choose multiple Locations.|
+|multiple_items_limit</br>`multipleItemsLimit`|number|no|Maximum number of items with configuration `multiple: true`.|
+|root_location_id</br>`rootLocationId`|number|no|UDW will display Locations only below this Content Tree element.|
+|starting_location_id</br>`startingLocationId`|number|no|This Location will be displayed as a starting Location in UDW.|
+|containers_only</br>`containersOnly`|true</br>false|no|When set to `true` only containers can be selected.|
+|allowed_content_types</br>`allowedContentTypes`|null</br>[]</br>[‘contentTypeIdentifier’]|yes|List of allowed Content Types:</br>`null` – all Content Types are allowed,</br>`[]` – empty table, no Content Types are allowed.|
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
|allowed_content_types</br>`allowedContentTypes`|null</br>[]</br>[`contentTypeIdentifier`]|yes|List of allowed Content Types:</br>`null` – all Content Types are allowed,</br>`[]` – empty table, no Content Types are allowed.|
```

File: docs/guide/extending/extending_udw.md [hunk 8]
```diff
@@ -33,120 +29,131 @@ const props = {
...
+|active_sort_clause</br>`activeSortClause`|DatePublished</br>ContentName|no|Clause under which children in the Content Tree will be sorted.|
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
|active_sort_clause</br>`activeSortClause`|DatePublished</br>ContentName|no|Sort Clause by which children in the Content Tree will be sorted.|
```

File: docs/guide/extending/extending_udw.md [hunk 9]
```diff
@@ -33,120 +29,131 @@ const props = {
...
+|active_tab</br>`activeTab`|browse</br>search</br>bookmarks|no|Starting tab in the UDW.|
+|active_view</br>`activeView`|finder</br>grid|no|Starting view in the UDW.|
+|allow_content_edit</br>`allowContentEdit`|true</br>false|yes|Visibility of `Edit` content button in UDW.|
+|selected_locations</br>`selectedLocations`|[]</br>[locationId]|no|Location that will be selected automatically.|
+
+### Content on the Fly Group
+
+|YML</br>React props|Values|Required|Definition|
+|-------------------|------|--------|----------|
+|allowed_languages</br>`allowedLanguages`|null</br>[]</br>[languageCode]|yes|Languages available in content on the fly: null - all, [] - none.|
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
|allowed_languages</br>`allowedLanguages`|null</br>[]</br>[languageCode]|yes|Languages available in Content on the Fly:</br>`null` - all,</br>`[]` - none.|
```

File: docs/guide/extending/extending_udw.md [hunk 10]
```diff
@@ -33,120 +29,131 @@ const props = {
...
+|allowed_locations</br>`allowedLocations`|null</br>[]</br>[locationId]|yes|Under which Location creating content is allowed:</br>null - everywhere, [] - nowhere.|
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
|allowed_locations</br>`allowedLocations`|null</br>[]</br>[locationId]|yes|Location under which creating content is allowed:</br>`null` - everywhere,</br>`[]` - nowhere.|
```

File: docs/guide/extending/extending_udw.md [hunk 11]
```diff
@@ -33,120 +29,131 @@ const props = {
...
+|preselected_language</br>`preselectedLanguage`|null</br>languageCode|yes|First language on the Content on the Fly language list:</br>null - language order defined in the system.|
+|preselected_content_type</br>`preselectedContentType`|null</br>contentTypeIdentifier|yes|Content marked in Content on the Fly.|
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
|preselected_content_type</br>`preselectedContentType`|null</br>contentTypeIdentifier|yes|Content selected in Content on the Fly.|
```

File: docs/guide/extending/extending_udw.md [hunk 12]
```diff
@@ -33,120 +29,131 @@ const props = {
...
+|hidden</br>`hidden`|true</br>false|yes|Content on the Fly visibility.|
+|auto_confirm_after_publish</br>`autoConfirmAfterPublish`|true</br>false|yes|If set to `true` UDW will be automatically closed after publishing the content.|
+
+### Tabs Config Group
+
+Configuration for each tab e.g. browse, search, bookmarks etc.
```
Comment (DominikaK) [inline docs/guide/extending/extending_udw.md]:
```suggestion
Configuration for all tabs, e.g. browse, search, bookmarks etc.
```

Comment (MagdalenaZuba) [reply docs/guide/extending/extending_udw.md]:
```suggestion
General configuration for tabs e.g. browse, search, bookmarks etc.
```

**Pull request merged**